components/org.wso2.carbon.identity.data.publisher.audit.idp.properties/src/main/java/org/wso2/carbon/identity/data/publisher/audit/idp/properties/model/IdPProperties.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.data.publisher.audit.idp.properties.model;
+
+import org.wso2.carbon.identity.base.IdentityRuntimeException;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * POJO for storing IdP properties.
+ */
+public class IdPProperties<T1 extends Object, T2 extends Object> {
+    private String idPName;
+    private String tenantDomain;
+    private Map<String, String> properties;
+    private String actionHolder;
+    private long timestamp;
+    private Map<T1, T2> parameters = new HashMap<>();
+
+    public String getIdPName() {
+        return idPName;
+    }
+
+    public void setIdPName(String idPName) {
+        this.idPName = idPName;
+    }
+
+    public String getTenantDomain() {
+        return tenantDomain;
+    }
+
+    public void setTenantDomain(String tenantDomain) {
+        this.tenantDomain = tenantDomain;
+    }
+
+    public Map<String, String> getProperties() {
+        return properties;
+    }
+
+    public void setProperties(Map<String, String> properties) {
+        this.properties = properties;
+    }
+
+    public String getActionHolder() {
+        return actionHolder;
+    }
+
+    public void setActionHolder(String actionHolder) {
+        this.actionHolder = actionHolder;
+    }
+
+    public long getTimestamp() {
+        return timestamp;
+    }
+
+    public void setTimestamp(long timestamp) {
+        this.timestamp = timestamp;
+    }
+
+    public void addParameter(T1 key, T2 value) {
+        if (this.parameters.containsKey(key)) {
+            throw IdentityRuntimeException.error("Parameters map trying to override existing key " +
+                    key);
+        }
```

Comment:
Is it necessary to throw this error. Does this mean you want to restrict overriding and existing parameter ?
What is the exact expectation ?

Reply:
This is done so in all the data publishers. I think it should be thrown.